Bottom: 9f58af7efb89068eb1a5b03f8916c657d4d6fa38
Top:    705088426da77fbc663958121737d38f9848d623
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 23:18:51 +0000

Refresh of hid-gerber-tidy-up-use_gc-to-m

---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index d45bcea..c49f456 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -916,6 +916,9 @@ gerber_set_draw_xor (hidGC gc, int xor_)
 static void
 use_gc (hidGC gc, int radius)
 {
+  if (f == NULL)
+    return;
+
   if (radius)
     {
       radius *= 2;
@@ -924,7 +927,7 @@ use_gc (hidGC gc, int radius)
 	  Aperture *aptr = findAperture (curr_aptr_list, radius, ROUND);
 	  if (aptr == NULL)
 	    pcb_fprintf (stderr, "error: aperture for radius %$mS type ROUND is null\n", radius);
-	  else if (f && !is_drill)
+	  else if (!is_drill)
 	    fprintf (f, "G54D%d*", aptr->dCode);
 	  linewidth = radius;
 	  lastcap = Round_Cap;
@@ -952,7 +955,7 @@ use_gc (hidGC gc, int radius)
       if (aptr == NULL)
         pcb_fprintf (stderr, "error: aperture for width %$mS type %s is null\n",
                  linewidth, shape == ROUND ? "ROUND" : "SQUARE");
-      else if (f)
+      else
 	fprintf (f, "G54D%d*", aptr->dCode);
     }
 }
